### ISSUE TITLE
loose interaction pairs when I compared two conditions 

### ISSUE THREAD

**Issue by Letichat** [sqjin/CellChat#553]
Hi,

Thanks for this package. It is really great and I am really happy to use it. I am currently analysing a dataset that contains wt and mutant cells. I first run cellChat on the whole dataset and found some interesting pairs. I then wanted to compare between Wt and mutant so I splited my Seurat object based on genotype and run cellChat individually on each dataset and merge them as suggested on the tutorial. However for one pair I am interested, I loose the interaction in mutant dataset (still detected in wt). However when I looked the data exp on Seurat I can still see receptor/ligand expression in the mutant dataset. I try to run on all the dataset but using different idents in WT and mutant (celltype1_wt, celltype_mutant, celltype2_wt, celltype2_mutant.....) and I can then find the interaction again in mutant. I guess it change the average exp or what is found with identifyOverExpressedGenes function. However I am not sure that it is really the good way to do it.

 Is there is a way to merge cellchat object prior to run the workflow? Or to detect more pairs on the split dataset? I already used the computeCommunProb(object, type = "truncatedMean" and trim = 0.1). I tried with trim =0 but it it did not change.

Thank for your advice.

**Comment by sqjin** [comment]
@Letichat It is likely that this signaling is not highly expressed in mutant dataset. Can you check the expression via ``computeAveExpr(cellchat, features = c("CXCL12","CXCR4"), type =  "truncatedMean", trim = 0.1)`` in the mutant dataset

**Comment by Letichat** [comment]
Here are the output for the ligand (the receptor is fine I guess because I can detect interaction with other ligands)

> computeAveExpr(cellchat.WT, features = c("vegfaa"), type =  "truncatedMean", trim = 0.1)
       EC PV+ neuron dl1 neuron Neuron 2       qERG v0 Neuron Neuron 1 v1 Neuron Oligodendrocytes NPC v2 Neuron         RP MN 2  OPC        FP MN 1         p3 dl neuron
vegfaa  0          0          0        0 0.05691057         0        0         0                0   0         0 0.08695652    0 0.16 0.5384615    0 0.03571429         0

> computeAveExpr(cellchat.flt1, features = c("vegfaa"), type =  "truncatedMean", trim = 0.1)
       EC PV+ neuron dl1 neuron  Neuron 2        qERG v0 Neuron Neuron 1 v1 Neuron Oligodendrocytes NPC v2 Neuron        RP MN 2 OPC  FP MN 1        p3 dl neuron
vegfaa  0          0          0 0.1111111 0.009708738         0   0.0625         0                0   0         0 0.2631579    0 0.2 0.2    0 0.1363636         0

For what I can see, there is still some expression here. 

I found that if I used a 0.1 threshold for identifyOverExpressedGenes, I could find my interaction again. The interaction are then similar to what I have on my whole dataset with different Idents for WT and mutant.

**Comment by Letichat** [comment]
Yes. Do you think it is OKay to do that? For what I can see I don't get non specific things as the pathway are the same as the ones I found when I run cellchat on the whole dataset.
